src/main/kotlin/Personnummer.kt

```diff
@@ -69,6 +65,8 @@ class Personnummer {
         return (luhnValue == control) && validDate(year, month, day)
     }
 
+    private fun parseYear(year : String) : String = if (year.length == 4) year.substring(2) else year
```

Comment:
This function could return an `Int` instead.

Thus `year = y.toInt()` could be replaced by:
```kotlin
year = parseYear(matchResult.groupValues[2])
```